v0/destinations/hs/util.js

```diff
@@ -300,6 +302,142 @@ const getEventAndPropertiesFromConfig = (message, destination, payload) => {
   return payload;
 };
 
+const splitEventsForCreateUpdate = async (inputs, destination) => {
+  const { Config } = destination;
+  const values = [];
+  const updateHubspotIds = [];
+  const firstMessage = inputs[0].message;
+  let objectType = null;
+  let identifierType = null;
+  if (firstMessage) {
+    objectType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .objectType;
+    identifierType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .identifierType;
+  }
+  inputs.map(async input => {
+    const { message } = input;
+    const { destinationExternalId } = getDestinationExternalIDInfoForRetl(
+      message,
+      "HS"
+    );
+    values.push(destinationExternalId);
+  });
+  let requestData = {
```

Comment:
we can extract this into a new function. This function is doing more than just splitting input messages